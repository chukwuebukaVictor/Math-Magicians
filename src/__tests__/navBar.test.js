import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
// import renderer from 'react-test-renderer';
import NavBar from '../components/Navbar';

// it ('Checking if Navigation Bar renders correctly', () => {
//   const tree = renderer.create(<Router> <NavBar /></Router>).toJSON();
//   expect(tree).toMatchSnapshot();
// });

describe('Test for Navbar component', () => {
  test('The component should render', () => {
    const { asFragment } = render(
      <Router>
        <NavBar />
      </Router>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
