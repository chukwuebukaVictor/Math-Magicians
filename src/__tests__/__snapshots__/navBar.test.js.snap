// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header component The component should render 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <h1>
      Math Magician
    </h1>
    <ul>
      <a
        aria-current="page"
        class="links active"
        href="/"
      >
        Home
      </a>
      <a
        class="links"
        href="/calculator"
      >
        Calculator
      </a>
      <a
        class="links"
        href="/quote"
      >
        Qoute
      </a>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Test for Navbar component The component should render 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <h1>
      Math Magician
    </h1>
    <ul>
      <a
        aria-current="page"
        class="links active"
        href="/"
      >
        Home
      </a>
      <a
        class="links"
        href="/calculator"
      >
        Calculator
      </a>
      <a
        class="links"
        href="/quote"
      >
        Qoute
      </a>
    </ul>
  </nav>
</DocumentFragment>
`;
